### PR TITLE
.Here API

### LINKED ISSUE #105: .here API
As you implement the API compile a document that helps us to understand what the API can do so that when we run into problems they become easier to fix.

### PR DESCRIPTION
**Pull request by ayesha1**
Closes #105 and #111

### CODE REVIEW

File: BayPass/APIs/GoogleMaps.swift [hunk 1]
```diff
@@ -88,18 +90,19 @@ class GoogleMaps {
...
-            let lineName = lineJson["short_name"] as? String ?? ""
+            let lineName = lineJson["name"] as? String ?? ""
+            if let agencies = lineJson["agencies"] as? [String: Any] {
+                let agencyName = agencies["name"] as? String?
+                ag = Agency(rawValue: agencyName as! String) ?? Agency.zero
+            }
 
             // TODO: This section relies on getting the fare prices from firebase and the line from the API first
-            let line = Line(name: lineName, code: 232, destination: "De Anza", stops: [])
+            let line = Line(name: lineName, agency: ag, destination: "De Anza", color: #colorLiteral(red: 0.2901960784, green: 0.5647058824, blue: 0.8862745098, alpha: 1), transitMode: TransitMode.bus)
```
Comment (timroesner) [inline BayPass/APIs/GoogleMaps.swift]: Why are you using a static Line here? I imagine that we can get the Line from the name which we use as key in the Map you created with the initial API calls?

File: BayPass/APIs/Here.swift [hunk 2]
```diff
@@ -0,0 +1,276 @@
...
+//
+//  Created by Ayesha Khan on 2/27/19.
+//  Copyright © 2019 Tim Roesner. All rights reserved.
+//
+
+import Alamofire
+import Foundation
+import MapKit
+
+class Here {
```
Comment (timroesner) [inline BayPass/APIs/Here.swift]: Did you decide on a data structure that fits our use case? I'd store it in a Singleton and initialize it with all stations and lines from the AppDelegate

File: BayPass.xcodeproj/xcshareddata/xcschemes/BayPass.xcscheme [hunk 3]
```diff
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <Scheme
-   LastUpgradeVersion = "1000"
+   LastUpgradeVersion = "1010"
```
Comment (timroesner) [inline BayPass.xcodeproj/xcshareddata/xcschemes/BayPass.xcscheme]: This could be an issue

File: BayPass.xcodeproj/project.pbxproj [hunk 4]
```diff
@@ -7,8 +7,6 @@
 	objects = {
 
 /* Begin PBXBuildFile section */
-		10DF627C28E33A8338F39522 /* Pods_BayPass.framework in Frameworks */ = {isa = PBXBuildFile; fileRef = BC737C8C1C489644ABDDFE78 /* Pods_BayPass.framework */; };
```
Comment (timroesner) [inline BayPass.xcodeproj/project.pbxproj]: Also slightly concerned with this deletion which could lead to errors in the tests

File: BayPass/Model/Map/Agency.swift [hunk 5]
```diff
@@ -5,21 +5,70 @@
...
+    case ACTransit = "AC " // AC Transit
+    case VTA = "SC" // VTA
+    case BART = "BAR" // Bart
+    case CalTrain = "CT" // Caltrain
+    case GoldenGateTransit = "GGT"
+    case Muni = "SF" // San Francisco Municipal Transportation Agency
+    case UnionCity = "UC" // Union City Transit
+    case ACE = "99" // Altamont Corridor Express
+    case SolTrans = "247" // SolsTrans
+    case SamsTrans = "sam" // SamsTrans
```
Comment (timroesner) [inline BayPass/Model/Map/Agency.swift]: Please change to "SamTrans" same goes for "SolTrans" these agencies do not have an additional 's'

Comment (ayesha1) [reply BayPass/Model/Map/Agency.swift]: but "short_name": "sam", and "name": "SolTrans" is "short_name": "247", from the API

Comment (ayesha1) [reply BayPass/Model/Map/Agency.swift]: Actually, nevermind! I thought you were talking about the string value

File: BayPass/APIs/Here.swift [hunk 6]
```diff
@@ -0,0 +1,312 @@
...
+    static let shared = Here()
+    private init() {}
+
+    // MARK: Get Requests for Here
+
+    // Returns an agency for given Station ID
+    func getAgencyFromStationId(stationId: Int, completion: @escaping (Agency) -> Void) {
```
Comment (timroesner) [inline BayPass/APIs/Here.swift]: What is the difference between this and ...

File: BayPass/APIs/Here.swift [hunk 7]
```diff
@@ -0,0 +1,312 @@
...
+                }
+                completion(results)
+            } else {
+                print("Failed in getting Lines")
+                completion([])
+            }
+        }
+    }
+
+    func getAgency(stationId: Int, completion: @escaping (Agency) -> Void) {
```
Comment (timroesner) [inline BayPass/APIs/Here.swift]: this function?

**Pull request merged**